src/network/p2p/event_loop.rs

```diff
@@ -405,34 +405,41 @@ impl EventLoop {
 					trace!("Identify Error event. PeerId: {peer_id:?}. Error: {error:?}");
 				},
 			},
-			SwarmEvent::Behaviour(BehaviourEvent::Mdns(event)) => match event {
-				mdns::Event::Discovered(addrs_list) => {
-					for (peer_id, multiaddr) in addrs_list {
-						trace!("MDNS got peer with ID: {peer_id:#?} and Address: {multiaddr:#?}");
-						self.swarm
-							.behaviour_mut()
-							.kademlia
-							.add_address(&peer_id, multiaddr);
-					}
-				},
-				mdns::Event::Expired(addrs_list) => {
-					for (peer_id, multiaddr) in addrs_list {
-						trace!("MDNS got expired peer with ID: {peer_id:#?} and Address: {multiaddr:#?}");
-
-						if self
-							.swarm
-							.behaviour_mut()
-							.mdns
-							.discovered_nodes()
-							.any(|&p| p == peer_id)
-						{
-							self.swarm
+			SwarmEvent::Behaviour(BehaviourEvent::Mdns(event)) => {
+				match event {
+					mdns::Event::Discovered(addrs_list) => {
+						addrs_list
+							.into_iter()
+							.filter(|(peer_id, multiaddr)| {
+								multiaddr
+									.to_string()
+									.contains(Protocol::P2p(*peer_id).tag())
+							})
+							.for_each(|(peer_id, multiaddr)| {
+								trace!("MDNS got peer with ID: {peer_id:#?} and Address: {multiaddr:#?}");
+
+								let _ = self.swarm.dial(multiaddr);
```

Comment:
I'd suggest that we don't use a dial here, as it would create a ton of traffic between peers on the initial boot.

What we could do is:

1. Abstract the Identify logic that deals with the version/mode checks
2. Use the same logic here as well as on the Identify 
3. Make that new check function testable

Reply:
This won't be doable because of the input that MDNS handler receives.

Another approach is to push the Identify from the handler to force it to trigger.